src/darwin/Framework/CHIP/templates/MTRCommandPayloadsObjc-src.zapt

```diff
@@ -58,6 +65,48 @@ NS_ASSUME_NONNULL_BEGIN
 {{/if}}
 {{/zcl_command_arguments}}
 
+{{#if (isStrEqual source "server")}}
+- (nullable instancetype)initWithResponseValue:(NSDictionary<NSString *, id> *)responseValue
+                                         error:(NSError * __autoreleasing *)error
+{
+  if (!(self = [super init])) {
+    return nil;
+  }
+
+  using DecodableType = chip::app::Clusters::{{asUpperCamelCase parent.name}}::Commands::{{asUpperCamelCase name}}::DecodableType;
+  chip::System::PacketBufferHandle buffer = [MTRBaseDevice _responseDataForCommand:responseValue
```

Comment:
Could we hang on to the raw response buffer (in a separate dictionary key) instead of re-encoding to TLV and then decoding again differently?

Reply:
We are not initially handed a buffer. We are initially handed a TLVReader positioned on the right thing.  We could try to do something clever with `GetReadPoint()` and `GetRemainingLength()`, but I have low confidence it would always do the right thing, and seems like an abstraction violation.

But also: encoding ourselves means we know the encoding will be valid TLV (we hope!).  So if the original TLV parse is in one process, then the dictionary gets shipped to another process, and we do this conversion in that other process, we are not parsing the from-the-network TLV in the process the dictionary got shipped to.

Longer-term I am considering having something where we don't go through TLV at all and instead go directly from the dictionary form to the typed form.  But this was easy to do, and gets us to an API that we can then improve under the hood.